openquake/hazardlib/gsim/idriss_2014.py

```diff
@@ -0,0 +1,183 @@
+# The Hazard Library
+# Copyright (C) 2015, GEM Foundation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Module exports :class:`Idriss2014`,
+"""
+from __future__ import division
+
+import numpy as np
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, SA
+
+
+class Idriss2014(GMPE):
+    """
+    Implements GMPE developed by Idriss 2014 and published as "An NGA-West2
+    Empirical Model for Estimating the Horizontal Spectral Values Generated
+    by Shallow Crustal Earthquakes.
+    (2014, Earthquake Spectra, Volume 30, No. 3, pages 1155 - 1177).
+
+    Idriss (2014) defines the GMPE only for the case in which Vs30 >= 450 m/s.
+    In the present implementation no check is made for the use of this model
+    for sites with Vs30 < 450 m/s
```

Comment:
I am not sure. The problem is that it is necessary to check if sites.vs30 < 450 on each call to the GMPE as it may not always apply depending on whether or not such sites are within the integration distance. Running the extra check seems like a little bit too much overhead to justify when the problem really belongs to the modeller.